pkgs/http_profile/lib/http_profile.dart

```diff
@@ -2,8 +2,186 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:async' show StreamController, StreamSink;
+import 'dart:developer'
+    show addHttpClientProfilingData, getHttpClientProfilingData, Timeline;
 import 'dart:io';
 
+/// Describes an event that occurred while issuing an HTTP request.
+class HttpProfileRequestEvent {
+  /// The time at which the event occurred, in microseconds.
+  final int timestamp;
```

Comment:
Is it maybe worth making this a `DateTime`? That way we can handle the conversion without developers possibly needing to pass around an integer.

Reply:
The timestamps returned by the service extensions currently use the timeline's monotonic clock. We could keep returning those kinds of timestamps from the service extensions and use `DateTime`s in this package if we add some conversion logic to the service extension code. Do you think we should do that?

Reply:
Ah that's a good point, I forgot we used the timeline's clock. In that case, if we stick with the current implementation, we should be very careful to mention this in documentation and guide developers to use [`Timeline.now`](https://api.flutter.dev/flutter/dart-developer/Timeline/now.html) to get this timestamp as close to the request as possible.

We _could_ switch this to using `DateTime` in the package and figure out how to convert to the monotonic clock in the service extension, but at that point we might just be better serializing the `DateTime` itself and just using that instead. I probably used the monotonic clock in the first place because we were scraping all this data from the timeline, but since we're not doing that anymore and we're doing this conversion in DevTools, it might be a good time to simplify this.

Reply:
Switching everything to use `DateTime` sounds good to me.